### ISSUE TITLE
paddle ocr 设置use_tensorrt=True 导致初始化卡住

### ISSUE THREAD

**Issue by caishenzhishou** [PaddlePaddle/Paddle#68503]
### 请提出你的问题 Please ask your question

执行初始化的时候 predictor = inference.create_predictor(config)这步会卡住
配置如下：

系统 | CUDA | CUDNN | TensorRT | python | paddlepaddle | PyTorch
-- | -- | -- | -- | -- | -- | --
windows11专业版显卡RTX2060 | 12.6 | 8.9.7.29 | 8.6.1.6 | 3.11 | paddlepaddle-gpu==3.0.0b1选CU12.3 | 2.4.1版本CU12.4
windows11专业版显卡RTX2060 | 11.8 | 8.6.0.163 | 8.5.1.7 | 3.10 | paddlepaddle-gpu==3.0.0b1选CU11.8 | 2.4.1版本CU11.8

入口代码：paddleocr = PaddleOCR(lang='ch', show_log=True, use_gpu=True, enable_mkldnn=True, use_tensorrt=True, use_angle_cls=False)每次运行到这里就卡住。debug之后，具体到最上面那步卡死。               麻烦专家们给个提示或者建议？感谢！！

**Comment by caishenzhishou** [comment]
一共试了2套环境，都不行

**Comment by caishenzhishou** [comment]
日志如下：
[2024/09/27 21:11:55] ppocr DEBUG: Namespace(help='==SUPPRESS==', use_gpu=True, use_xpu=False, use_npu=False, use_mlu=False, ir_optim=True, use_tensorrt=True, min_subgraph_size=15, precision='fp32', gpu_mem=500, gpu_id=0, image_dir=None, page_num=0, det_algorithm='DB', det_model_dir='C:\\Users\\XCCG05A20016/.paddleocr/whl\\det\\ch\\ch_PP-OCRv4_det_infer', det_limit_side_len=960, det_limit_type='max', det_box_type='quad', det_db_thresh=0.3, det_db_box_thresh=0.6, det_db_unclip_ratio=1.5, max_batch_size=10, use_dilation=False, det_db_score_mode='fast', det_east_score_thresh=0.8, det_east_cover_thresh=0.1, det_east_nms_thresh=0.2, det_sast_score_thresh=0.5, det_sast_nms_thresh=0.2, det_pse_thresh=0, det_pse_box_thresh=0.85, det_pse_min_area=16, det_pse_scale=1, scales=[8, 16, 32], alpha=1.0, beta=1.0, fourier_degree=5, rec_algorithm='SVTR_LCNet', rec_model_dir='C:\\Users\\XCCG05A20016/.paddleocr/whl\\rec\\ch\\ch_PP-OCRv4_rec_infer', rec_image_inverse=True, rec_image_shape='3, 48, 320', rec_batch_num=6, max_text_length=25, rec_char_dict_path='D:\\software\\python310\\lib\\site-packages\\paddleocr\\ppocr\\utils\\ppocr_keys_v1.txt', use_space_char=True, vis_font_path='./doc/fonts/simfang.ttf', drop_score=0.5, e2e_algorithm='PGNet', e2e_model_dir=None, e2e_limit_side_len=768, e2e_limit_type='max', e2e_pgnet_score_thresh=0.5, e2e_char_dict_path='./ppocr/utils/ic15_dict.txt', e2e_pgnet_valid_set='totaltext', e2e_pgnet_mode='fast', use_angle_cls=False, cls_model_dir='C:\\Users\\XCCG05A20016/.paddleocr/whl\\cls\\ch_ppocr_mobile_v2.0_cls_infer', cls_image_shape='3, 48, 192', label_list=['0', '180'], cls_batch_num=6, cls_thresh=0.9, enable_mkldnn=True, cpu_threads=10, use_pdserving=False, warmup=False, sr_model_dir=None, sr_image_shape='3, 32, 128', sr_batch_num=1, draw_img_save_dir='./inference_results', save_crop_res=False, crop_res_save_dir='./output', use_mp=False, total_process_num=1, process_id=0, benchmark=False, save_log_path='./log_output/', show_log=True, use_onnx=False, return_word_box=False, output='./output', table_max_len=488, table_algorithm='TableAttn', table_model_dir=None, merge_no_span_structure=True, table_char_dict_path=None, layout_model_dir=None, layout_dict_path=None, layout_score_threshold=0.5, layout_nms_threshold=0.5, kie_algorithm='LayoutXLM', ser_model_dir=None, re_model_dir=None, use_visual_backbone=True, ser_dict_path='../train_data/XFUND/class_list_xfun.txt', ocr_order_method=None, mode='structure', image_orientation=False, layout=True, table=True, ocr=True, recovery=False, use_pdf2docx_api=False, invert=False, binarize=False, alphacolor=(255, 255, 255), lang='ch', det=True, rec=True, type='ocr', savefile=False, ocr_version='PP-OCRv4', structure_version='PP-StructureV2', use_cuda=True)
[2024/09/27 21:30:23] ppocr WARNING: Since the angle classifier is not initialized, it will not be used during the forward process
[2024/09/27 21:30:23] ppocr DEBUG: dt_boxes num : 72, elapsed : 0.11554646492004395
E0927 21:30:24.357795 19416 helper.h:131] 3: [builder.cpp::nvinfer1::builder::Builder::~Builder::307] Error Code 3: API Usage Error (Parameter check failed at: builder.cpp::nvinfer1::builder::Builder::~Builder::307, condition: mObjectCounter.use_count() == 1. Destroying a builder object before destroying objects it created leads to undefined behavior.
)

**Comment by caishenzhishou** [comment]
pip list 显示如下：nvidia-cublas-cu11       11.11.3.6
nvidia-cuda-nvrtc-cu11   11.8.89
nvidia-cuda-runtime-cu11 11.8.89
nvidia-cudnn-cu11        8.9.4.19
nvidia-cufft-cu11        10.9.0.58
nvidia-curand-cu11       10.3.0.86
nvidia-cusolver-cu11     11.4.1.48
nvidia-cusparse-cu11     11.7.5.86
tensorrt                 8.5.1.7

是不是这里的CUDNN版本不对？但是我都是按照步骤执行安装的。

**Comment by vivienfanghuagood** [comment]
@cuicheng01  麻烦指导一下用户如何在PaddleOCR中打开inference的GLOG日志，以便排查问题，谢谢！

**Comment by caishenzhishou** [comment]
@BiynXu @vivienfanghuagood @jzhang533 @wanglun 几位大佬，目前我这边安装了CUDA 11.8、CUDNN 8.6.0.163、tensorRT8.5.3.1 windows环境下通过wsl安装了ubuntu 部署，首次识别很快，第二次再识别就卡机了。请问问题在哪？

**Comment by caishenzhishou** [comment]
日志如下：
WARNING: Logging before InitGoogleLogging() is written to STDERR
I1010 06:45:16.760123  1052 dynamic_loader.cc:176] Set paddle lib path : /home/xinxuan818/.local/lib/python3.10/site-packages/paddle/libs
I1010 06:45:17.088349  1052 init.cc:100] Before Parse: argc is 2, Init commandline: dummy --tryfromenv=save_static_runtime_data,dataloader_use_file_descriptor,gpugraph_enable_segment_merge_grads,mklml_dir,cpu_deterministic,print_allocator_trace_info,gpugraph_debug_gpu_memory,use_cuda_malloc_async_allocator,set_to_1d,cudnn_dir,sync_after_alloc,manually_trans_conv_filter,alloc_fill_value,graph_load_in_parallel,communicator_is_sgd_optimizer,query_dest_rank_by_multi_node,jit_engine_type,gpugraph_offload_gather_copy_maxsize,enable_async_trace,sync_nccl_allreduce,gpugraph_offload_param_stat,use_auto_growth_v2,disable_dyshape_in_train,ir_inplace_kernel_blacklist,cuda_dir,enable_dump_main_program,enable_dependency_builder_debug_info,enable_neighbor_list_use_uva,gemm_use_half_precision_compute_type,prim_check_ops,gpugraph_hbm_table_load_factor,conv2d_disable_cudnn,rocksdb_path,fraction_of_cuda_pinned_memory_to_use,new_executor_serial_run,gpugraph_merge_grads_segment_size,free_idle_chunk,eager_delete_tensor_gb,new_executor_use_local_scope,search_cache_max_number,graph_embedding_split_infer_mode,cudnn_exhaustive_search,enable_pir_in_executor_trace_run,log_memory_stats,tracer_onednn_ops_off,enable_exit_when_partial_worker,add_dependency_for_communication_op,check_nan_inf_level,async_trace_count,gpugraph_force_device_batch_num_equal,use_mkldnn,print_ir,rpc_send_thread_num,enable_auto_detect_gpu_topo,enable_cse_in_dy2st,fleet_executor_with_standalone,graph_edges_split_debug,benchmark_nccl,tracer_profile_fname,enable_gpu_memory_usage_log_mb,nvidia_package_dir,init_allocated_mem,all_blocks_convert_trt,free_when_no_cache_hit,reader_queue_speed_test_mode,cuda_memory_async_pool_realease_threshold,graph_metapath_split_opt,auto_free_cudagraph_allocations_on_launch,static_runtime_data_save_path,tracer_onednn_ops_on,gpugraph_storage_mode,reallocate_gpu_memory_in_mb,convert_all_blocks,logging_trunc_pir_py_code,run_kp_kernel,allocator_strategy,communicator_max_merge_var_num,prim_enable_dynamic,enable_cublas_tensor_op_math,gpugraph_slot_feasign_max_num,use_pinned_memory,enable_cinn_accuracy_check,new_executor_static_build,enable_adjust_op_order,op_dir,deny_cinn_ops,low_precision_op_list,static_executor_perfstat_filepath,enable_api_kernel_fallback,use_cinn,nccl_dir,gpugraph_sparse_table_storage_mode,use_system_allocator,sort_sum_gradient,fuse_parameter_groups_size,check_infer_symbolic,gpugraph_load_node_list_into_hbm,gpugraph_dedup_pull_push_mode,max_inplace_grad_add,enable_auto_rdma_trans,graph_edges_split_only_by_src_id,mkl_dir,auto_growth_chunk_size_in_mb,cusolver_dir,dynamic_static_unified_comm,win_cuda_bin_dir,enable_sparse_inner_gather,prim_backward,graph_edges_split_mode,cusparselt_dir,check_kernel_launch,use_auto_growth_pinned_allocator,memory_fraction_of_eager_deletion,gpu_memory_limit_mb,use_autotune,enable_opt_get_features,trt_ibuilder_cache,einsum_opt,host_trace_level,gpugraph_enable_gpu_direct_access,graph_edges_debug_node_num,enable_record_memory,fraction_of_gpu_memory_to_use,gpugraph_parallel_stream_num,gpugraph_enable_hbm_table_collision_stat,prim_forward_blacklist,prim_forward,get_host_by_name_time,nccl_blocking_wait,new_executor_use_cuda_graph,cse_max_count,gpugraph_offload_param_extends,initial_cpu_memory_in_mb,pir_subgraph_saving_dir,executor_log_deps_every_microseconds,cinn_compile_thread_num,cupti_dir,gpugraph_parallel_copyer_split_maxsize,check_nan_inf,gpu_allocator_retry_time,graph_get_neighbor_id,cache_inference_while_scope,allreduce_record_one_event,enable_unused_var_check,fuse_parameter_memory_size,cusparse_dir,multiple_of_cupti_buffer_size,embedding_deterministic,initial_gpu_memory_in_mb,enable_gpu_memory_usage_log,cublaslt_exhaustive_search_times,enable_interpretercore_launch_cinn,cudnn_exhaustive_search_times,pir_apply_shape_optimization_pass,fast_eager_deletion_mode,custom_device_mem_record,prim_enabled,gpugraph_enable_print_op_debug,communicator_send_queue_size,pir_debug,enable_all2all_use_fp16,cudnn_batchnorm_spatial_persistent,npu_storage_format,new_executor_use_inplace,benchmark,tensorrt_dir,dygraph_debug,lapack_dir,selected_gpus,inner_op_parallelism,call_stack_level,prim_all,use_stream_safe_cuda_allocator,new_executor_sequential_run,local_exe_sub_scope_limit,logging_pir_py_code_dir,use_virtual_memory_auto_growth,allow_cinn_ops,pe_profile_fname,enable_tracker_all2all,use_fast_math,conv_workspace_size_limit,enable_fuse_parallel_matmul_pass,paddle_num_threads,print_sub_graph_dir,use_cuda_managed_memory,enable_pir_api,enable_cinn_compile_cache,graph_neighbor_size_percent,use_stride_kernel,graph_edges_debug_node_id,pir_apply_inplace_pass,cinn_subgraph_graphviz_dir,multi_node_sample_use_gpu_table,prim_skip_dynamic,curand_dir,use_shm_cache,cublas_dir,cudnn_deterministic,apply_pass_to_program,fraction_of_cpu_memory_to_use,dist_threadpool_size,enable_graph_multi_node_sampling,tensor_operants_mode,enable_pir_in_executor,enable_pir_with_pt_in_dy2st,enable_cinn_auto_tune,eager_delete_scope
I1010 06:45:17.088462  1052 init.cc:108] After Parse: argc is 2
I1010 06:45:17.088557  1052 os_info.cc:117] SetCurrentThreadName MainThread
I1010 06:45:17.342315  1052 pybind.cc:2374] Initialize tensor operants successfully
I1010 06:45:17.434238  1052 amp_auto_cast.cc:114] -- The size of all_ops: 1170 --
I1010 06:45:17.434288  1052 amp_auto_cast.cc:115] -- The size of supported_ops: 208 --
I1010 06:45:17.434293  1052 amp_auto_cast.cc:116] -- The size of unsupported_ops: 966 --
I1010 06:45:17.617311  1052 global_value_getter_setter.cc:191] set FLAGS_nvidia_package_dir to /home/xinxuan818/.local/lib/python3.10/site-packages/paddle/../nvidia
I1010 06:45:17.617372  1052 global_value_getter_setter.cc:191] set FLAGS_cublas_dir to /home/xinxuan818/.local/lib/python3.10/site-packages/paddle/../nvidia/cublas/lib
I1010 06:45:17.617380  1052 global_value_getter_setter.cc:191] set FLAGS_cudnn_dir to /home/xinxuan818/.local/lib/python3.10/site-packages/paddle/../nvidia/cudnn/lib
I1010 06:45:17.617386  1052 global_value_getter_setter.cc:191] set FLAGS_curand_dir to /home/xinxuan818/.local/lib/python3.10/site-packages/paddle/../nvidia/curand/lib
I1010 06:45:17.617393  1052 global_value_getter_setter.cc:191] set FLAGS_cusolver_dir to /home/xinxuan818/.local/lib/python3.10/site-packages/paddle/../nvidia/cusolver/lib
I1010 06:45:17.617398  1052 global_value_getter_setter.cc:191] set FLAGS_cusparse_dir to /home/xinxuan818/.local/lib/python3.10/site-packages/paddle/../nvidia/cusparse/lib
I1010 06:45:17.617403  1052 global_value_getter_setter.cc:191] set FLAGS_nccl_dir to /home/xinxuan818/.local/lib/python3.10/site-packages/paddle/../nvidia/nccl/lib
I1010 06:45:17.617408  1052 global_value_getter_setter.cc:191] set FLAGS_cupti_dir to /home/xinxuan818/.local/lib/python3.10/site-packages/paddle/../nvidia/cuda_cupti/lib
I1010 06:45:17.617691  1052 imperative.cc:701] Tracer(0x56124154f160) set expected place Place(gpu:0)
System check identified no issues (0 silenced).

October 10, 2024 - 07:49:16
Django version 5.1.2, using settings 'ocr_project.settings'
Starting development server at http://0.0.0.0:8000/
Quit the server with CONTROL-C.

当前获取到的锁的值：1
ocr/recognise/20240919/1.jpg
[2024/10/10 07:49:18] ppocr DEBUG: Namespace(help='==SUPPRESS==', use_gpu=True, use_xpu=False, use_npu=False, use_mlu=False, ir_optim=True, use_tensorrt=True, min_subgraph_size=15, precision='fp32', gpu_mem=500, gpu_id=0, image_dir=None, page_num=0, det_algorithm='DB', det_model_dir='/home/xinxuan818/.paddleocr/whl/det/ch/ch_PP-OCRv4_det_infer', det_limit_side_len=960, det_limit_type='max', det_box_type='quad', det_db_thresh=0.3, det_db_box_thresh=0.6, det_db_unclip_ratio=1.5, max_batch_size=10, use_dilation=False, det_db_score_mode='fast', det_east_score_thresh=0.8, det_east_cover_thresh=0.1, det_east_nms_thresh=0.2, det_sast_score_thresh=0.5, det_sast_nms_thresh=0.2, det_pse_thresh=0, det_pse_box_thresh=0.85, det_pse_min_area=16, det_pse_scale=1, scales=[8, 16, 32], alpha=1.0, beta=1.0, fourier_degree=5, rec_algorithm='SVTR_LCNet', rec_model_dir='/home/xinxuan818/.paddleocr/whl/rec/ch/ch_PP-OCRv4_rec_infer', rec_image_inverse=True, rec_image_shape='3, 48, 320', rec_batch_num=6, max_text_length=25, rec_char_dict_path='/home/xinxuan818/.local/lib/python3.10/site-packages/paddleocr/ppocr/utils/ppocr_keys_v1.txt', use_space_char=True, vis_font_path='./doc/fonts/simfang.ttf', drop_score=0.5, e2e_algorithm='PGNet', e2e_model_dir=None, e2e_limit_side_len=768, e2e_limit_type='max', e2e_pgnet_score_thresh=0.5, e2e_char_dict_path='./ppocr/utils/ic15_dict.txt', e2e_pgnet_valid_set='totaltext', e2e_pgnet_mode='fast', use_angle_cls=False, cls_model_dir='/home/xinxuan818/.paddleocr/whl/cls/ch_ppocr_mobile_v2.0_cls_infer', cls_image_shape='3, 48, 192', label_list=['0', '180'], cls_batch_num=6, cls_thresh=0.9, enable_mkldnn=True, cpu_threads=10, use_pdserving=False, warmup=False, sr_model_dir=None, sr_image_shape='3, 32, 128', sr_batch_num=1, draw_img_save_dir='./inference_results', save_crop_res=False, crop_res_save_dir='./output', use_mp=False, total_process_num=1, process_id=0, benchmark=False, save_log_path='./log_output/', show_log=True, use_onnx=False, return_word_box=False, output='./output', table_max_len=488, table_algorithm='TableAttn', table_model_dir=None, merge_no_span_structure=True, table_char_dict_path=None, layout_model_dir=None, layout_dict_path=None, layout_score_threshold=0.5, layout_nms_threshold=0.5, kie_algorithm='LayoutXLM', ser_model_dir=None, re_model_dir=None, use_visual_backbone=True, ser_dict_path='../train_data/XFUND/class_list_xfun.txt', ocr_order_method=None, mode='structure', image_orientation=False, layout=True, table=True, ocr=True, recovery=False, use_pdf2docx_api=False, invert=False, binarize=False, alphacolor=(255, 255, 255), lang='ch', det=True, rec=True, type='ocr', savefile=False, ocr_version='PP-OCRv4', structure_version='PP-StructureV2')

**Comment by caishenzhishou** [comment]
我发现每次识别之后需要手动清除.paddleocr目录下的缓存才可以，否则再次调用就会卡机，请问这里如何优化？
